lib/openxml/has_properties.rb

```diff
@@ -0,0 +1,85 @@
+module OpenXml
+  module HasProperties
+
+    def self.included(base)
+      base.extend ClassMethods
+    end
+
+    module ClassMethods
+      def properties_tag(*args)
+        @properties_tag = args.first if args.any?
+        @properties_tag
+      end
+
+      def value_property(name, as: nil)
+        attr_reader name
+
+        properties[name] = (as || name).to_s
+
+        class_eval <<-CODE, __FILE__, __LINE__ + 1
+        def #{name}=(value)
+          property_key = "#{name}".to_sym
+          class_name = properties[property_key].split("_").map(&:capitalize).join
+          prop_module = class.name.split("::")[0...-2].push("Properties").join("::").constantize
```

Comment:
Is `.constantize` from ActiveSupport? (we don't list that as a runtime dependency of this library — and I'd like not to)

Reply:
D'oh! I don't know if I realized that was from ActiveSupport. I'll refactor it away...